language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/signature/SignatureInfoModelBuilder.java

```diff
@@ -0,0 +1,342 @@
+    }
+
+    private SignatureInformation getSignatureInfo(String label, List<ParameterInformation> parameters) {
+        SignatureInformation expandedSignatureInfo = new SignatureInformation();
+        expandedSignatureInfo.setDocumentation(this.signatureDescription);
+        expandedSignatureInfo.setLabel(label);
+        expandedSignatureInfo.setParameters(parameters);
+        return expandedSignatureInfo;
+    }
+
+    private void fillParamInfoModels() {
+        fillDescriptionModels();
+        Optional<FunctionTypeSymbol> functionTypeSymbol = getFunctionType(symbol);
+        List<ParameterSymbol> parameterSymbols = functionTypeSymbol
+                .map(typeSymbol -> typeSymbol.params().orElse(new ArrayList<>()))
+                .orElse(Collections.emptyList());
+        if (parameterSymbols.isEmpty()) {
+            return;
+        }
+
+        List<Parameter> parameters = parameterSymbols
+                .subList(skipFirstParam() ? 1 : 0, parameterSymbols.size())
+                .stream()
+                .map(param -> new Parameter(param, false, false, context))
+                .collect(Collectors.toList());
+
+        Optional<ParameterSymbol> restParam = functionTypeSymbol.flatMap(FunctionTypeSymbol::restParam);
+        restParam.ifPresent(parameter -> parameters.add(new Parameter(parameter, false, true, context)));
+
+        // Create a list of param info models
+        for (Parameter param : parameters) {
+            String desc = param.getName().isPresent() && paramsMap.containsKey(param.getName().get()) ?
+                    paramsMap.get(param.getName().get()) : "";
+            parameterModels.add(new ParameterInfoModel(param, desc));
+        }
+
+        firstIncludedRecordParam = this.parameterModels.stream()
+                .filter(ParameterInfoModel::isIncludedRecordParam)
+                .findFirst()
+                .orElse(null);
+    }
+
+    private boolean skipFirstParam() {
+        return (symbol.kind() == SymbolKind.FUNCTION || symbol.kind() == SymbolKind.METHOD)
+                && (symbol.getModule().isPresent() && CommonUtil.isLangLib(symbol.getModule().get().id()));
+    }
+
+    private void fillDescriptionModels() {
+        Optional<Documentation> documentation =
+                symbol instanceof Documentable ? ((Documentable) symbol).documentation() : Optional.empty();
+        // Check for documentations of the function and parameters
+        if (documentation.isEmpty()) {
+            return;
+        }
+        if (documentation.get().description().isPresent()) {
+            this.setSignatureDescription(documentation.get().description().get().trim(), context);
+        }
+        documentation.get().parameterMap().forEach(paramsMap::put);
+    }
+
+    private void setSignatureDescription(String signatureDescription, SignatureContext signatureContext) {
+        SignatureInformationCapabilities capabilities = signatureContext.capabilities().getSignatureInformation();
```

Comment:
Shall we check if the` signatureContext.capabilities() != null` to be on the safe side

Reply:
This is a usage of an old context builder API. This has been fixed by making sure the capabilities are set by default.